### PR TITLE
Disable UDP Send Queuing (Temporarily)

### PR DESCRIPTION
**Pull request by nibanks**
Updates the UDP send logic to only queue to the worker thread if it's not actively doing something else. This is an attempt to stabilize perf. Note, it's possible this introduces reordering, which I have to think about.

UPDATE: Adds a build flag to control the UDP send queuing logic in general. Right now, disabling it because we have some other perf instability issues we need to debug before we can re-enable this.

### CODE REVIEW

File: src/platform/datapath_winuser.c [hunk 1]
```diff
@@ -357,6 +357,11 @@ typedef struct CXPLAT_DATAPATH_PROC {
     //
     uint16_t Index;
 
+    //
+    // Flag indicates that the worker thread is currently actively processing.
+    //
+    uint16_t WorkerActive : 1;
```
Comment (ThadHouse) [inline src/platform/datapath_winuser.c:363]: Is there a reason to make this a unit16_t and a bitflag? Unless we plan on making operations with this interlocked.

Comment (nibanks) [reply src/platform/datapath_winuser.c:363]: Only a bit is needed. And there was space for a `uint16_t`.

**Comment by nibanks** [comment]
As is, this made perf drop even more. I think we need something a bit more complicated. Like start by always queuing the packets for a given connection to the UDP worker. If that worker ever gets to be too overloaded, then fallback to doing the rest inline for the lifetime of the connection. Still needs some experimentation.

**Comment by nibanks** [comment]
So, the only consistent changes I can see by disabling UDP send queuing are:

1. User mode throughput dropped back down to ~7 Gbps (from ~7.8 Gbps).
2. User mode, openssl (default) RPS went back up to ~1 Million RPS

I'm of mixed opinion for next steps. Part of me thinks that we should make this a build time flag or link it to TLS library (enable for schannel, but disable for openssl).

**Comment by maolson-msft** [comment]
Remind me: why were sends queued in the first place? Were they always queued by a worker thread [that was working on some other queued operation]?

**Comment by nibanks** [comment]
> Remind me: why were sends queued in the first place? Were they always queued by a worker thread [that was working on some other queued operation]?

We recently made did some experiments and made the change to queue the UDP sends to the datapath worker thread instead of running them inline on the QUIC worker thread, because of the CPU usage imbalance between the two. QUIC worker thread was maxed out at 100%, but the datapath worker was 60% to 80% usage. When we queued the UDP sends, we effectively moved 10% CPU usage from the QUIC worker to the UDP worker, achieving a better over all balance, and then more overall throughput (+10% throughput).

**Review by ThadHouse — APPROVED** [review]

**Review by anrossi — APPROVED** [review]

**Pull request merged**